setup/publish.sh

```diff
@@ -1,72 +1,177 @@
+    echo "]"
+    echo "then export"
+    echo "> export ADMIN_SECRET_KEY=<priv key diamond>"
+    exit 1
 fi
 
-publishedObjs=$(echo "$publish_res" | jq -r '.objectChanges[] | select(.type == "published")')
+# specific to blackjack contract only
+if [ -z "${BJ_PLAYER_SECRET_KEY}" ]; then
+    echo "Please setup env var BJ_PLAYER_SECRET_KEY with the secret key of the client address to play bj"
+    echo "E.g. find out position of bj player client address"
+    echo "> sui client addresses"
+    echo "heliotrope | 0xCAFE | *"
+    echo "diamond    | 0xFACE | "
+    echo "then get respective private key from sui keystore"
+    echo "> cat <PATH_TO_SUI_INSTALLATION>/.sui/sui_config/sui.keystore"
+    echo "[<priv key heliotrope>"
+    echo " <priv key diamond>"
+    echo "]"
+    echo "then export"
+    echo "> export BJ_PLAYER_SECRET_KEY=<priv key heliotrope>"
+    exit 1
+fi
 
-PACKAGE_ID=$(echo "$publishedObjs" | jq -r '.packageId')
+# setup network
+NETWORK=
+SUFFIX=
+
+case "$ENV" in
+"testnet")
+    NETWORK="https://fullnode.testnet.sui.io:443"
+    ;;
+"devnet")
+    NETWORK="https://fullnode.devnet.sui.io:443"
+    ;;
+"local")
+    NETWORK="http://localhost:9000"
+    SUFFIX=".local"
+    ;;
+*)
+    echo "Unknown env $ENV"
+    exit 1 
+esac
+
+# Switch to client to publish
+echo "Switching to client: ${ADMIN_ADDRESS}"
+sui client switch --address ${ADMIN_ADDRESS}
+
+# Switch to selected env
+echo "Switching to env: {$ENV}"
+sui client switch --env $ENV
+
+if [ ${DRY_RUN} == true ]; then
+    echo "Done - not publishing"
+    exit 0
+fi
 
-newObjs=$(echo "$publish_res" | jq -r '.objectChanges[] | select(.type == "created")')
+# Do actual puslish
+echo "Do actual publish"
+PUBLISH_RES=$(sui client publish --skip-fetch-latest-git-deps --json ${MOVE_PACKAGE_PATH})
 
-HOUSE_ADMIN_CAP=$(echo "$newObjs" | jq -r 'select (.objectType | contains("single_player_blackjack::HouseAdminCap")).objectId')
+echo ${PUBLISH_RES} > .publish.res.json
 
-suffix=""
-if [ $# -eq 0 ]; then
-  suffix=".localnet"
+# Check if the command succeeded (exit status 0) and for success in text
+if [[ "$PUBLISH_RES" =~ "error" && "$PUBLISH_RES" != *"Success"* ]]; then
+    # If yes, print the error message and exit the script
+    echo "Error during move contract publishing. Details : $PUBLISH_RES"
+    exit 1
 fi
 
+# Publish success
+echo "Publish successful"
+echo "Creating new env variables"
+PUBLISHED_OBJS=$(echo "$PUBLISH_RES" | jq -r '.objectChanges[] | select(.type == "published")')
+PACKAGE_ID=$(echo "$PUBLISHED_OBJS" | jq -r '.packageId')
+CREATED_OBJS=$(echo "$PUBLISH_RES" | jq -r '.objectChanges[] | select(.type == "created")')
+HOUSE_ADMIN_CAP=$(echo "$CREATED_OBJS" | jq -r 'select (.objectType | contains("single_player_blackjack::HouseAdminCap")).objectId')
+
+echo "Publish new env var to setup/.env: "
+echo "SUI_NETWORK=$NETWORK"
+echo "PACKAGE_ADDRESS=$PACKAGE_ID"
+echo "ADMIN_ADDRESS=$ADMIN_ADDRESS"
+echo "ADMIN_SECRET_KEY=$ADMIN_SECRET_KEY"
```

Comment:
I believe we should avoid printing the secret keys.
Same thinking as in the following comment

Reply:
resolved in latest commit